src/main/java/org/veupathdb/service/eda/ss/model/variable/StringVariable.java

```diff
@@ -7,26 +7,37 @@
 import org.veupathdb.service.eda.ss.model.variable.binary.ByteArrayConverter;
 
 public class StringVariable extends VariableWithValues<byte[]> {
-  private final static int LARGEST_STRING = 200;
+  private StringBinaryProperties binaryProperties;
 
-  public StringVariable(Variable.Properties varProperties, VariableWithValues.Properties valueProperties) {
+  public StringVariable(Variable.Properties varProperties,
+                        VariableWithValues.Properties valueProperties,
+                        StringBinaryProperties binaryProperties) {
     super(varProperties, valueProperties);
     validateType(VariableType.STRING);
+    this.binaryProperties = binaryProperties;
   }
   
   // static version for use when we don't have an instance
-  public static BinaryConverter<byte[]> getGenericBinaryConverter() {
-    return new ByteArrayConverter(LARGEST_STRING);
+  public static BinaryConverter<byte[]> getGenericBinaryConverter(BinaryProperties binaryProperties) {
```

Comment:
Convert argument here to StringBinaryProperties so you don't have to cast on 22.

Reply:
This was a bit tricky because of the usage from our binary file printer. Wasn't able to come up with a great solution